pkg/cli/admin/mustgather/summary.go

```diff
@@ -0,0 +1,179 @@
+		availableMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorProgressing); condition != nil {
+		progressingMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorDegraded); condition != nil {
+		degradedMessage = condition.Message
+	}
+	if condition := v1helpers.FindStatusCondition(clusterOperator.Status.Conditions, configv1.OperatorUpgradeable); condition != nil {
+		upgradeableMessage = condition.Message
+	}
+
+	switch {
+	case !available:
+		return fmt.Sprintf("clusteroperator/%s is not available (%v) because %v", clusterOperator.Name, availableMessage, degradedMessage)
+	case degraded:
+		return fmt.Sprintf("clusteroperator/%s is degraded because %v", clusterOperator.Name, degradedMessage)
+	case !upgradeable:
+		return fmt.Sprintf("clusteroperator/%s is not upgradeable because %v", clusterOperator.Name, upgradeableMessage)
+	case progressing:
+		return fmt.Sprintf("clusteroperator/%s is progressing: %v", clusterOperator.Name, progressingMessage)
+	case available && !progressing && !degraded && upgradeable:
+		return ""
+	default:
+		return fmt.Sprintf("clusteroperator/%s is in an edge case", clusterOperator.Name)
+	}
+}
+
+func humanSummaryForClusterVersion(clusterVersion *configv1.ClusterVersion) string {
+	if clusterVersion == nil {
+		return "Cluster is in a version state we don't recognize"
+	}
+
+	isInstalling :=
+		len(clusterVersion.Status.History) == 0 ||
+			(len(clusterVersion.Status.History) == 1 && clusterVersion.Status.History[0].State != configv1.CompletedUpdate)
+	isUpdating := len(clusterVersion.Status.History) > 1 && clusterVersion.Status.History[0].State != configv1.CompletedUpdate
+	isStable := len(clusterVersion.Status.History) > 0 && clusterVersion.Status.History[0].State == configv1.CompletedUpdate
+
+	lastChangeHumanDuration := "<unknown>"
+	if len(clusterVersion.Status.History) > 0 {
+		lastChangeHumanDuration = units.HumanDuration(time.Now().Sub(clusterVersion.Status.History[0].StartedTime.Time))
+	}
+
+	progressingConditionMessage := "<unknown>"
+	for _, condition := range clusterVersion.Status.Conditions {
+		if condition.Type == "Progressing" {
+			progressingConditionMessage = condition.Message
+			break
+		}
+	}
+	switch {
+	case isInstalling:
+		return fmt.Sprintf("Installing %q for %v: %v",
+			clusterVersion.Status.Desired.Version, lastChangeHumanDuration, progressingConditionMessage)
+	case isUpdating:
+		return fmt.Sprintf("Updating to %q from %q for %v: %v",
+			clusterVersion.Status.Desired.Version, clusterVersion.Status.History[1].Version, lastChangeHumanDuration, progressingConditionMessage)
+	case isStable:
+		return fmt.Sprintf("Stable at %q", clusterVersion.Status.History[0].Version)
+	default:
+		return fmt.Sprintf("Unknown state")
```

Comment:
I make a comment above about 'default' states, and providing 'guidance on next steps', can we also address that here.

Reply:
> I make a comment above about 'default' states, and providing 'guidance on next steps', can we also address that here.

I don't know why it would be in a state like this.  But someone else can work it out and potentially add something if it is common enough.